storage/src/test/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTaskTest.java

```diff
@@ -284,20 +279,19 @@ public void testCanReprocessSkippedRecords() throws InterruptedException {
         consumer.updateEndOffsets(Collections.singletonMap(toRemoteLogPartition(anotherMetadataPartition), 0L));
         final Set<TopicIdPartition> assignments = Collections.singleton(tpId0);
         consumerTask.addAssignmentsForPartitions(assignments);
-        thread.start();
-        TestUtils.waitForCondition(() -> consumerTask.isUserPartitionAssigned(tpId0), "Timed out waiting for " + tpId0 + " to be assigned");
+        consumerTask.ingestRecords();
 
         // Adding metadata records in the order opposite to the order of assignments
         addRecord(consumer, metadataPartition, tpId1, 0);
         addRecord(consumer, metadataPartition, tpId0, 1);
-        TestUtils.waitForCondition(() -> consumerTask.readOffsetForMetadataPartition(metadataPartition).equals(Optional.of(1L)), "Couldn't read record");
+        consumerTask.ingestRecords();
+        assertTrue(() -> consumerTask.readOffsetForMetadataPartition(metadataPartition).equals(Optional.of(1L)), "Couldn't read record");
         // Only one record is processed, tpId1 record is skipped as unassigned
         // but read offset is 1 e.g., record for tpId1 has been read by consumer
         assertEquals(1, handler.metadataCounter);
 
         // Adding assignment for tpId1 after related metadata records have already been read
         consumerTask.addAssignmentsForPartitions(Collections.singleton(tpId1));
```

Comment:
In this test, we used doAnswer to mock out the consumer behavior, because of the multithreading. I don't think this mocking is necessary anymore now that there aren't any spurious poll() calls in-between the two addAssignmentsForPartition calls.

You can remove doAnswer in this test, and the spy on the consumer in beforeEach.